Update Kotlin and KotlinX.Serialization version

Version 0.4.0 of this library crashed with Kotlin version 1.3.70 because https://github.com/Kotlin/kotlinx.serialization/issues/736
Please, update Kotlin version to 1.3.70 and KotlinX.Serialization version to 0.20.0.

I got this crash. Might be related, but I don't know if it's on this library or on kotlinx.serialization.

```
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at kotlinx.serialization.SerializationKt.invokeSerializerGetter(Serialization.kt:37)
        at kotlinx.serialization.SerializationKt.compiledSerializer(Serialization.kt:11)
        at kotlinx.serialization.PlatformUtilsKt.serializer(PlatformUtils.kt:11)
        at kotlinx.serialization.JvmResolvingKt.serializerByTypeToken(JvmResolving.kt:61)
        at kotlinx.serialization.JvmResolvingKt.serializerByTypeToken(JvmResolving.kt:84)
        at com.jakewharton.retrofit2.converter.kotlinx.serialization.Factory.responseBodyConverter(Factory.kt:23)
        at retrofit2.Retrofit.nextResponseBodyConverter(Retrofit.java:352)
        at retrofit2.Retrofit.responseBodyConverter(Retrofit.java:335)
        at retrofit2.HttpServiceMethod.createResponseConverter(HttpServiceMethod.java:113)
        at retrofit2.HttpServiceMethod.parseAnnotations(HttpServiceMethod.java:82)
        at retrofit2.ServiceMethod.parseAnnotations(ServiceMethod.java:37)
2020-03-05 22:28:51.176 6842-6842/com.saladevs.bytelift E/AndroidRuntime:     at retrofit2.Retrofit.loadServiceMethod(Retrofit.java:192)
        at retrofit2.Retrofit$1.invoke(Retrofit.java:149)
        at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
        	... 16 more
     Caused by: java.lang.NoSuchMethodError: No direct method <init>(Ljava/lang/String;Lkotlinx/serialization/internal/GeneratedSerializer;I)V in class Lkotlinx/serialization/internal/SerialClassDescImpl; or its super classes (declaration of 'kotlinx.serialization.internal.SerialClassDescImpl' appears in /data/app/com.saladevs.bytelift-x1pCW_pzXi5TImI6mGNzlA==/base.apk)
        at ***.Build$$serializer.<clinit>(Build.kt:12)
        at ***.Build$Companion.serializer(Unknown Source:0)
        	... 31 more
```

--- a/build.gradle
+++ b/build.gradle
@@ -2,8 +2,8 @@ buildscript {
   ext.versions = [
     'kotlin': '1.3.70',
     'serializationRuntime': '0.20.0',
-    'retrofit': '2.7.2',
-    'okhttp': '3.12.10',
+    'retrofit': '2.6.4',
+    'okhttp': '3.14.7',
     'junit': '4.13',
     'truth': '1.0',
   ]

--- a/build.gradle
+++ b/build.gradle
@@ -1,6 +1,11 @@
 buildscript {
   ext.versions = [
-    'kotlin': '1.3.60'
+    'kotlin': '1.3.70',
+    'serializationRuntime': '0.20.0',
+    'retrofit': '2.7.2',
+    'okhttp': '3.12.10',
+    'junit': '4.13',
+    'truth': '1.0',
   ]
   repositories {
     mavenCentral()
@@ -25,11 +30,12 @@ repositories {
 }
 
 dependencies {
-  api 'com.squareup.retrofit2:retrofit:2.5.0'
+  api "com.squareup.retrofit2:retrofit:${versions.retrofit}"
   api "org.jetbrains.kotlin:kotlin-stdlib:${versions.kotlin}"
-  api 'org.jetbrains.kotlinx:kotlinx-serialization-runtime:0.14.0'
+  api "org.jetbrains.kotlinx:kotlinx-serialization-runtime:${versions.serializationRuntime}"
 
-  testImplementation 'junit:junit:4.12'
-  testImplementation 'com.squareup.okhttp3:mockwebserver:3.11.0'
-  testImplementation 'com.google.truth:truth:0.42'
+  testImplementation "junit:junit:${versions.junit}"
+  testImplementation "com.squareup.okhttp3:mockwebserver:${versions.okhttp}"
+  testImplementation "com.google.truth:truth:${versions.truth}"
+  testImplementation "org.jetbrains.kotlinx:kotlinx-serialization-protobuf:${versions.serializationRuntime}"
 }

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.2.2-bin.zip
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/src/test/java/com/jakewharton/retrofit2/converter/kotlinx/serialization/KotlinSerializationConverterFactoryBytesTest.kt
+++ b/src/test/java/com/jakewharton/retrofit2/converter/kotlinx/serialization/KotlinSerializationConverterFactoryBytesTest.kt
@@ -1,6 +1,6 @@
 package com.jakewharton.retrofit2.converter.kotlinx.serialization
 
-import kotlinx.serialization.SerialId
+import kotlinx.serialization.protobuf.ProtoId
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.protobuf.ProtoBuf
 import okhttp3.MediaType
@@ -31,7 +31,7 @@ class KotlinSerializationConverterFactoryBytesTest {
   }
 
   @Serializable
-  data class User(@SerialId(1) val name: String)
+  data class User(@ProtoId(1) val name: String)
 
   @Before fun setUp() {
     val contentType = MediaType.get("application/x-protobuf")

--- a/src/test/java/com/jakewharton/retrofit2/converter/kotlinx/serialization/KotlinSerializationConverterFactoryStringTest.kt
+++ b/src/test/java/com/jakewharton/retrofit2/converter/kotlinx/serialization/KotlinSerializationConverterFactoryStringTest.kt
@@ -1,6 +1,7 @@
 package com.jakewharton.retrofit2.converter.kotlinx.serialization
 
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.UnstableDefault
 import kotlinx.serialization.json.Json
 import okhttp3.MediaType
 import okhttp3.mockwebserver.MockResponse
@@ -28,6 +29,7 @@ class KotlinSerializationConverterFactoryStringTest {
   @Serializable
   data class User(val name: String)
 
+  @UnstableDefault
   @Before fun setUp() {
     val contentType = MediaType.get("application/json; charset=utf-8")
     val retrofit = Retrofit.Builder()

